src/dreyfus_index.erl

```diff
@@ -120,23 +120,34 @@ init({DbName, Index}) ->
 
 handle_call({await, RequestSeq}, From,
             #state{
-                index=#index{current_seq=Seq}=Index,
+                index=#index{dbname=DbName,name=IdxName,ddoc_id=DDocId,current_seq=Seq}=Index,
                 index_pid=IndexPid,
                 updater_pid=nil,
                 waiting_list=WaitList
             }=State) when RequestSeq > Seq ->
-    UpPid = spawn_link(fun() -> dreyfus_index_updater:update(IndexPid, Index) end),
-    {noreply, State#state{
-        updater_pid=UpPid,
-        waiting_list=[{From,RequestSeq}|WaitList]
-    }};
+    DbName2 = mem3:dbname(DbName),
+    <<"_design/", GroupId/binary>> = DDocId,
+    NewState = case dreyfus_util:in_black_list(DbName2, GroupId, IdxName) of
+        false ->
+            UpPid = spawn_link(fun() ->
+                dreyfus_index_updater:update(IndexPid,Index)
+            end),
+            State#state{
+                updater_pid=UpPid,
+                waiting_list=[{From,RequestSeq}|WaitList]
+            };
+        _ ->
+            couch_log:notice("Index Blocked from Updating - db: ~p, ddocid: ~p name: ~p", [DbName, DDocId, IdxName]),
```

Comment:
more than 80 characters

Reply:
It looks like all 3 arguments are strings maybe we should use `'~s'`? Not a big deal.